logprep/processor/dissecter/rule.py

```diff
@@ -0,0 +1,217 @@
+    :linenos:
+    :caption: Given dissecter rule
+
+    filter: message
+    dissecter:
+        mapping:
+            message: "%{}of %{extracted.message_float} and a int of %{extracted.message_int}"
+        convert_datatype:
+            extracted.message_int: "int"
+            extracted.message_float: "float"
+    description: '...'
+
+..  code-block:: json
+    :linenos:
+    :caption: Incomming event
+
+    {"message": "This message has a float of 1.23 and a int of 1337"}
+
+..  code-block:: json
+    :linenos:
+    :caption: Processed event
+
+    {
+        "message": "This message has a float of 1.23 and a int of 1337",
+        "extracted": {"message_float": 1.23, "message_int": 1337},
+    }
+
+Dissect Pattern Language
+------------------------
+
+The dissect pattern describes the textual format of the source field.
+
+Given a dissect pattern of :code:`%{field1} %{field2}` the source field value will be dissected into
+everything before the first whitespace which would be written into the field `field1` and everything
+after the first whitespace which would be written into the field `field2`.
+
+The string surrounded by :code:`%{}` is the desired target field. This can be declared in dotted
+field notation. (e.g. :code:`%{target.subfield1.subfield2}`). Every subfield between the first and
+the last subfield will be created.
+
+In default the target field will always be overwritten with the captured value. If you want to
+append to a preexisting target field value as string or list you have to use the :code:`+` operator.
+
+It is possible to capture the target field name from the source field value with the notation
+:code:`%{?<your name for the reference>}` (e.g. :code:`%{?key1}`). This can be referred to with the
+notation :code:`%{&<the reference>}` (e.g. :code:`%{&key1}`) afterwards in the same dissection
+pattern. References can be combined with the append operator.
+
+.. autoclass:: logprep.processor.dissecter.rule.DissecterRule.Config
+   :members:
+   :undoc-members:
+   :inherited-members:
+   :noindex:
+
+Examples for dissection and datatype conversion:
+------------------------------------------------
+
+.. datatemplate:import-module:: tests.unit.processor.dissecter.test_dissecter
+   :template: testcase-renderer.tmpl
+
+"""
+from functools import partial
+import re
+from typing import Callable, List, Tuple
+from attrs import define, validators, field, Factory
+from logprep.processor.base.rule import Rule
+from logprep.filter.expression.filter_expression import FilterExpression
+from logprep.processor.base.exceptions import InvalidRuleDefinitionError
+from logprep.util.helper import add_field_to, get_dotted_field_value
+
+DISSECT = r"(%\{[+&?]?[^%{]*\})"
+SEPERATOR = r"((?!%\{.*\}).+)"
+
+append_as_list = partial(add_field_to, extends_lists=True)
+
+
+def add_and_overwrite(event, target_field, content, *_):
+    """wrapper for add_field_to"""
+    add_field_to(event, target_field, content, overwrite_output_field=True)
```

Comment:
same

Reply:
I understand this argument, but I think keeping stuff together makes it also easier for other developers to know which functionality already exists. If someone only looks into the `helper.py` because they want to use the add_field_to method, they don't get the idea to check also if there is a similar method in the dissecter rule, so they might end up writing it again. Keeping similar things in the same location could prevent that.

Reply:
done